hillstone.py

```diff
@@ -257,12 +264,14 @@ def wait_network(self):
                 self.server_udp_port = int.from_bytes(res[Payload.SVR_UDP_PORT], byteorder='big')
                 self.ip_ipv4 = ipaddress.IPv4Interface((res[Payload.CLT_PRIV_IPV4], network.prefixlen))
                 self.gateway_ipv4 = ipaddress.IPv4Address(res[Payload.SVR_PRIV_IPV4])
-                self.dns_ipv4 = ipaddress.IPv4Address(res[Payload.DNS_IPV4])
+                self.dns_ipv4 = ipaddress.IPv4Address(self.Int2IP(int.from_bytes(res[Payload.DNS_IPV4],'big')))
```

Comment:
I think we should simply truncate the `res[Payload.DNS_IPV4]` to `[:4]`. Converting to string is not necessary, because the following `IPV4Address(...)` will immediately parse it to numbers.